### DISCUSSION TITLE
Cannot Publish Draft Release

### DISCUSSION THREAD

**Discussion by TaffarelJr** [community/community#16500] [Actions]
I'm using GitHub Actions to run my CI/CD pipeline. At the end of the workflow, it uses Cake and GitReleaseManager to create a draft Release in GitHub and upload the build artifacts to it. A human can then inspect the Release, and publish it if they so choose.

Everything seems to be in order in the Release: the name of the tag (to be created) is correct, the Release title matches it, release notes generated properly, and the assets are attached. But when I click `Publish release`, it gives me an error at the top:

`We weren't able to create the release for you. Make sure you have a valid tag.`

I've tried clearing the tag name, re-typing it, selecting it from the dropdown, ... it seems to be completely broken. What is it that could be preventing me from publishing this release?

Screenshot:
![image](https://user-images.githubusercontent.com/5290662/168226842-8ebe87fd-f674-48fe-8a16-5a6aded8c8b9.png)

**Comment by jamilbk** [comment]
Facing the same issue. Having this unexpectedly break is **not fun** when it comes to release day and all the rest of the team is ready to launch...

Need more descriptive errors.

**Reply by jamilbk** [reply]
Turns out it was our tag protection rules. I guess publishing a release follows the same tag protection rules as pushing? Working now but would be better if the error was a little more clear. Otherwise thanks Github for an awesome product!

**Reply by airtower-luna** [reply]
Making a release involves creating a tag unless the release uses an existing tag, so that makes sense. Thanks for sharing what you figured out! :smiley_cat:

**Comment by TaffarelJr** [comment]
Update: The issue suddenly went away for me a few weeks back. Not sure what changed--I didn't modify anything in my release process. I assumed GitHub fixed it on their own, perhaps without knowing about this thread.

If you're still experiencing it, I'd suggest putting in a support ticket.
